File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size() - 1;
        int ans = nums[0];

        while (l <= r) {
            if (nums[l] < nums[r]) {
                return min(ans, nums[l]);
            }
            int m = (r - l)/2 + l;

            ans = min(ans, nums[m]);
            if(nums[m] >= nums[l]) {
                l = m + 1;
            } else {
                r = m - 1;
            }

        }

        return ans;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Hashmap method
        group_map = defaultdict(int)
        ans = 0

        for num in nums:
            if not group_map[num]:
                group_map[num] = group_map[num-1] + 1 + group_map[num+1]
                group_map[num - group_map[num - 1]] = group_map[num]
                group_map[num + group_map[num + 1]] = group_map[num]
                ans = max(ans, group_map[num])

        return ans